### PR TITLE
feat: use eslint-import-resolver-typescript instead

### LINKED ISSUE #58: Use `eslint-import-resolver-typescript` instead of `eslint-import-resolver-node`?
https://github.com/alexgorbatchev/eslint-import-resolver-typescript

### PR DESCRIPTION
**Pull request by JounQin**
close #58

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -10,12 +11,13 @@ async function runEslint(string, config) {
 		overrideConfig: config,
 	});
 
-	const [firstResult] = await eslint.lintText(string, {filePath: '_x.ts'});
+	const [firstResult] = await eslint.lintText(string, {filePath: path.resolve('test', '_x.ts')});
 
 	return firstResult.messages;
 }
 
-test.failing('main', async t => {
```
Comment (JounQin) [inline test/test.js:18]: The previous test case is absolutely wrong.

File: package.json [hunk 2]
```diff
@@ -43,7 +43,11 @@
 		"hint",
 		"simple"
 	],
+	"dependencies": {
+		"eslint-import-resolver-typescript": "^2.7.1"
```
Comment (sindresorhus) [inline package.json:47]: Shareable ESLint configs cannot have dependencies.

Comment (JounQin) [reply package.json:47]: Not sure to understand, or should I list it in `peer`?

Comment (sindresorhus) [reply package.json:47]: https://github.com/eslint/eslint/issues/3458

Comment (JounQin) [reply package.json:47]: Interesting, but I'm also using this for a long time without any issue.

https://github.com/1stG/configs/blob/master/packages/eslint-config/package.json#L30

Comment (stefan-toubia) [reply package.json:47]: FWIW I have a shared ESLint config with dependencies and it's only plugins that are a problem, everything else works as expected.

```json
    "@typescript-eslint/eslint-plugin": "^4.28.3",
    "@typescript-eslint/parser": "^4.28.3",
    "babel-eslint": "^10.1.0",
    "eslint": "^7.18.0",
    "eslint-config-airbnb": "^18.2.0",
    "eslint-config-airbnb-typescript": "^12.3.1",
    "eslint-config-prettier": "^6.12.0",
    "eslint-import-resolver-alias": "^1.1.2",
    "eslint-import-resolver-typescript": "^2.3.0",
    "eslint-import-resolver-webpack": "^0.13.2",
    "eslint-plugin-babel": "^5.3.0",
    "eslint-plugin-import": "^2.22.0",
    "eslint-plugin-jsx-a11y": "^6.3.1",
    "eslint-plugin-react": "^7.20.3",
    "eslint-plugin-react-hooks": "^4.0.8"
```

Comment (sindresorhus) [reply package.json:47]: I would still prefer to wait until we move to flat ESLint config, which should have no problem with any kind of dependency.

**Pull request closed without merging**